### PR TITLE
Fix #1304: use simplejson instead of json in JSONEnvelope

### LINKED ISSUE #1304: non-JSON output of out-of-range float values (nan, inf, -inf)
## Context

The Python handler returns invalid response body (non-JSON) in case the inference result dict contains `nan`, `inf` or `-inf` values. This is due to Python's `json` encoder actually **not** returning valid JSON in these cases, unless the `allow_nan=False` argument is passed when calling `json.dumps`.

* torchserve version: 0.4.2

## Expected Behavior
When the handler returns a Python dict such as this `{"number": nan}`, the response should be this JSON: `{"number": null}`.

## Current Behavior
The response is  `{"number": Nan}`, which is not JSON.

## Possible Solution
Add `allow_nan=False` argument to the `json.dumps` call at https://github.com/pytorch/serve/blob/master/ts/torch_handler/request_envelope/json.py#L73.

## Steps to Reproduce
1. Override the BaseHandler to return a dict containing out-of-range float values.
2. Run a request
3. Observe non-JSON responses

### PR DESCRIPTION
**Pull request by vvekic**
## Description

Fixes #1304 by serializing with `simplejson` and its `ignore_nan` argument, which properly converts Python `nan`, `inf` and `-inf` values to JSON `null`. `simplejson` is also added to the main requirements list.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

Failing test case for this issue wasn't present. It is added in this PR in `ts/torch_handler/unit_tests/test_envelopes.py`.
Confirmed passing by running `/workspaces/serve/ts/torch_handler/unit_tests/run_unit_tests.sh`.

### CODE REVIEW

**Comment by msaroufim** [comment]
Very cool! PR looks fine overall, let me try it out and keep you posted

File: requirements/common.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ future
 psutil
 wheel
 requests
+simplejson
```
Comment (msaroufim) [inline requirements/common.txt:5]: let's just double check that other requirements.txt also need this or not and whether json is used anywhere else in the codebase

File: ts/torch_handler/unit_tests/test_envelopes.py [hunk 2]
```diff
@@ -46,6 +56,17 @@ def test_json_batch(handle_fn, model_context):
     results = envelope.handle(test_data, model_context)
     assert(results == expected_result)
 
+def test_json_nan(identity_handle_fn, identity_model_context):
+    test_data = [{'body':{
+        'instances': [[2.0, float("NaN")], [5.0, 4.0]]
```
Comment (msaroufim) [inline ts/torch_handler/unit_tests/test_envelopes.py:61]: mind also adding a test for `inf`? you can put it here

**Review by msaroufim — CHANGES_REQUESTED** [review]

File: requirements/common.txt (hunk 1 above)
Comment (vvekic) [reply requirements/common.txt:5]: i wasn't sure where to put this, so i chose common.txt. it's unlikely simpljson is used elsewhere. if that's the case - where do you propose to move this?

**Review by msaroufim — APPROVED** [review]

**Comment by msaroufim** [comment]
@lxning what do you think, should we merge this? LGTM

**Comment by maaquib** [comment]
@vvekic Thanks for this
We should just make sure there isn't any significant perf difference with using `simplejson` with larger payloads

**Comment by msaroufim** [comment]
While I do appreciate the benefits that come with better data imputation, my concern with using non standard JSON packages is performance and long term maintenance. So for now will close this PR and we can revisit if you feel strongly about it

**Pull request closed without merging**